File: download_assets.py
import urllib.request
import os

# Create directory
os.makedirs('public/branding', exist_ok=True)

# Supabase details
project_id = 'nxlabwiewewwkwemtvfj' # Extracted from the Header.tsx url
base_url = f"https://{project_id}.supabase.co/storage/v1/object/public/branding"

try:
    urllib.request.urlretrieve(f"{base_url}/logo.png", "public/branding/logo.png")
    print("Downloaded logo.png")
except Exception as e:
    print("Failed to download logo.png:", e)
    # try public:logo.png?
    try:
        urllib.request.urlretrieve(f"{base_url}/public:logo.png", "public/branding/logo.png")
        print("Downloaded public:logo.png")
    except Exception as e2:
        print("Failed to download public:logo.png:", e2)


try:
    urllib.request.urlretrieve(f"{base_url}/timbre.png", "public/branding/timbre.png")
    print("Downloaded timbre.png")
except Exception as e:
    print("Failed to download timbre.png:", e)

File: download_assets.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');

const env = fs.readFileSync('.env.local', 'utf8');
let supabaseUrl = '';
let supabaseKey = '';
env.split('\n').forEach(line => {
  if (line.startsWith('NEXT_PUBLIC_SUPABASE_URL=')) supabaseUrl = line.split('=')[1].trim();
  if (line.startsWith('NEXT_PUBLIC_SUPABASE_ANON_KEY=')) supabaseKey = line.split('=')[1].trim();
});

const supabase = createClient(supabaseUrl, supabaseKey);

async function download() {
  const { data: logoData, error: logoErr } = await supabase.storage.from('branding').download('logo.png');
  if (logoData) {
    const buffer = Buffer.from(await logoData.arrayBuffer());
    fs.writeFileSync('public/branding/logo.png', buffer);
    console.log('logo.png downloaded');
  } else {
    console.error('Error logo:', logoErr);
  }

  const { data: timbreData, error: timbreErr } = await supabase.storage.from('branding').download('timbre.png');
  if (timbreData) {
    const buffer = Buffer.from(await timbreData.arrayBuffer());
    fs.writeFileSync('public/branding/timbre.png', buffer);
    console.log('timbre.png downloaded');
  } else {
    console.error('Error timbre:', timbreErr);
  }
}

download();
